[Term Entry] JavaScript Array.prototype.some() (#2609)
Codecademy/docs: content/javascript/concepts/arrays/terms/some/some.md

---
Title: '.some()'
Description: 'Run a conditional through an array and returns a boolean if any value fulfills the conditional.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The **`.some()`** method tests whether at least one item in the array passes the test implemented by the provided function. It returns true if any item in the array succeeds the test, otherwise it reuturns false.

## Syntax

```pseudo
array.some(callback);
```

- `callback`: A function that takes in an element as a parameter. The element is then determined if it passes or fails the test.

## Example

> Pass in a callback function as a conditional check.

Reviewer: Be a little more specific about what each example is doing.

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 2 === 0

console.log(numbers.some(callback))
```

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 6 === 0

console.log(numbers.some(callback))
```

## Codebyte Example

> Below are several examples of running the some method through the `numbers` array:

Reviewer: ```suggestion
Below are several examples of running the `.some()` method through the `numbers` array:
```

---
Title: '.some()'
Description: 'Run a conditional through an array and returns a boolean if any value fulfills the conditional.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The **`.some()`** method tests whether at least one element in the array passes the test implemented by the provided function. It returns true if, in the array, it finds an element for which the provided function returns true; otherwise it returns false. It doesn't modify the array.

## Syntax

> ```js

Reviewer 2: ```suggestion
```pseudo
```

array.some(callback);
```

- `callback`: A function that takes in an element as a parameter. The element is then determined if it passes or fails the test.

## Example

Pass in a callback function as a conditional check.

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 2 === 0

console.log(numbers.some(callback))
// Returns true
```

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 6 === 0

console.log(numbers.some(callback))
// Returns false
```

> ### Codebyte Example

Reviewer 2: ```suggestion
## Codebyte Example
```

Below are several examples of running the some method through the `numbers` array:

```codebyte/js
let numbers=[1, 2, 3, 4, 5, 6];

> // Returns true

Reviewer 2: ```suggestion
```
I don't think there's any need to add comments in codebytes as the user can run and check the out themselves

let callback = (element) => element % 3 === 0;
let isDivisibleByThree = numbers.some(callback);
console.log(isDivisibleByThree);

> // Returns false

Reviewer 2: ```suggestion
```

Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The **`.some()`** method tests whether at least one item in the array passes the test implemented by the provided function. It returns true if any item in the array succeeds the test, otherwise it reuturns false.

## Syntax

```pseudo
array.some(callback);
```

- `callback`: A function that takes in an element as a parameter. The element is then determined if it passes or fails the test.

## Example

Pass in a callback function as a conditional check. This will trigger an iteration process.
For each iteration, the current number is divided by 2 to determine whether the remainder will equal 0. Once the callback function returns a value, the iteration will move onto the next number in the `numbers` array.

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 2 === 0

console.log(numbers.some(callback))
```

This example results in the following output:

```shell
true
```

> 

Reviewer: There should be a sentence summarizing this second example.

---
Title: '.some()'
Description: 'Runs a conditional through an array and returns a boolean if any value fulfills the conditional.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The **`.some()`** method tests whether at least one item in the array passes the test implemented by the provided function. It returns true if any item in the array succeeds the test, otherwise it reuturns false.

## Syntax

```pseudo
array.some(callback);
```

- `callback`: A function that takes in an element as a parameter. The element is then determined if it passes or fails the test.

## Example

> Pass in a callback function as a conditional check. This will trigger an iteration process.

Reviewer 3: ```suggestion
```

> For each iteration, the current number is divided by 2 to determine whether the remainder will equal 0. Once the callback function returns a value, the iteration will move onto the next number in the `numbers` array.

Reviewer 3: ```suggestion
In the example below the `.some()` function is used to determine if any of the values in the array are even (divisible by 2). 
```

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 2 === 0

console.log(numbers.some(callback))
```

This example results in the following output:

```shell
true
```

> Pass in a callback function as a conditional check. This will trigger an iteration process.

Reviewer 3: I would recommend deleting this second example. All of these are very similar and the codebyte is essentially duplicating the first two examples. Ideally, every example is demonstrating something new about the method, concept, etc.

For each iteration, the current number is divided by 6 to determine whether the remainder will equal 0. Once the callback function returns a value, the iteration will move onto the next number in the `numbers` array.

```js
const numbers = [2, 4, 5, 7, 8];
const callback = (element) => element % 6 === 0

console.log(numbers.some(callback))
```

This example results in the following output:

```shell
false
```

## Codebyte Example

> Below are several examples of running the `.some()` method through the `numbers` array:

Reviewer 3: ```suggestion
Below are two examples of running the `.some()` function on the `numbers` array:
```

```codebyte/js
let numbers=[1, 2, 3, 4, 5, 6];

> let callback = (element) => element % 3 === 0;
> let isDivisibleByThree = numbers.some(callback);
> console.log(isDivisibleByThree);
> 
> let callback = (element) => element % 7 === 0;
> let isDivisibleBySeven = numbers.some(callback);

Reviewer 3: ```suggestion
let test1 = (element) => element % 3 === 0;
let isDivisibleByThree = numbers.some(test1);
console.log(isDivisibleByThree);

let test2 = (element) => element % 7 === 0;
let isDivisibleBySeven = numbers.some(test2);
```
